### PR TITLE
Warn on shallow clones

### LINKED ISSUE #4615: docfx crash report: System.NotSupportedException
# docfx crash report: System.NotSupportedException

docfx: `1.0.0`
x64: `True`
git: `git version 2.21.0.windows.1`

## repro steps

Run `E:\docfx\src\docfx\bin\Debug\netcoreapp2.2\docfx.dll build E:\docfx-v3-migration\.temp\AspNetCore.Docs\AspNetCore.Docs --legacy --verbose` in `E:\docfx\src\docfx\bin\Debug\netcoreapp2.2`

## callstack

```
System.NotSupportedException: Error computing git log '-3' for 'E:/docfx-v3-migration/.temp/AspNetCore.Docs/AspNetCore.Docs/'
   at Microsoft.Docs.Build.FileCommitProvider.LoadCommits(String committish) in E:\docfx\src\docfx\lib\git\FileCommitProvider.cs:line 244
   at Microsoft.Docs.Build.FileCommitProvider.<>c__DisplayClass12_1.<GetCommitHistory>b__5() in E:\docfx\src\docfx\lib\git\FileCommitProvider.cs:line 73
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location where exception was thrown ---
   at System.Lazy`1.CreateValue()
   at Microsoft.Docs.Build.FileCommitProvider.GetCommitHistory(String file, String committish) in E:\docfx\src\docfx\lib\git\FileCommitProvider.cs:line 71
   at Microsoft.Docs.Build.GitCommitProvider.GetCommitHistory(String fullPath, Repository repo, String committish) in E:\docfx\src\docfx\lib\git\GitCommitProvider.cs:line 34
   at Microsoft.Docs.Build.GitCommitProvider.GetCommitHistory(Document document, String committish) in E:\docfx\src\docfx\lib\git\GitCommitProvider.cs:line 15
   at Microsoft.Docs.Build.ContributionProvider.GetGitUrls(Document document) in E:\docfx\src\docfx\build\metadata\ContributionProvider.cs:line 165
   at Microsoft.Docs.Build.BuildPage.Build(Context context, Document file, TableOfContentsMap tocMap, Action`1 buildChild) in E:\docfx\src\docfx\build\page\BuildPage.cs:line 40
   at Microsoft.Docs.Build.Build.BuildFile(Context context, Document file, TableOfContentsMap tocMap, MonikerMap monikerMap, Action`1 buildChild) in E:\docfx\src\docfx\build\Build.cs:line 173
   at Microsoft.Docs.Build.Build.<>c__DisplayClass2_0.<<BuildFiles>g__BuildOneFile|3>d.MoveNext() in E:\docfx\src\docfx\build\Build.cs:line 125
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Docs.Build.Build.<>c__DisplayClass2_1.<<BuildFiles>b__0>d.MoveNext() in E:\docfx\src\docfx\build\Build.cs:line 98
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Docs.Build.ParallelUtility.<>c__DisplayClass3_0`1.<<ForEach>g__Run|0>d.MoveNext() in E:\docfx\src\docfx\lib\ParallelUtility.cs:line 123
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Docs.Build.ParallelUtility.ForEach[T](IEnumerable`1 source, Func`3 action, Func`2 predicate, Action`2 progress) in E:\docfx\src\docfx\lib\ParallelUtility.cs:line 101
   at Microsoft.Docs.Build.Build.BuildFiles(Context context, Docset docset, TableOfContentsMap tocMap) in E:\docfx\src\docfx\build\Build.cs:line 96
   at Microsoft.Docs.Build.Build.Run(String docsetPath, Repository repository, String locale, CommandLineOptions options, ErrorLog errorLog, RestoreMap restoreMap, Repository fallbackRepo) in E:\docfx\src\docfx\build\Build.cs:line 59
   at Microsoft.Docs.Build.Build.Run(String docsetPath, CommandLineOptions options, ErrorLog errorLog) in E:\docfx\src\docfx\build\Build.cs:line 26
   at Microsoft.Docs.Build.Docfx.Run(String[] args) in E:\docfx\src\docfx\cli\Docfx.cs:line 79
   at Microsoft.Docs.Build.Docfx.Main(String[] args) in E:\docfx\src\docfx\cli\Docfx.cs:line 23
```

## dotnet --info

```
.NET Core SDK (reflecting any global.json):
 Version:   2.2.202
 Commit:    8a7ff6789d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.2.202\

Host (useful for support):
  Version: 2.2.3
  Commit:  6b8ad509b6

.NET Core SDKs installed:
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.400 [C:\Program Files\dotnet\sdk]
  2.1.500 [C:\Program Files\dotnet\sdk]
  2.1.503 [C:\Program Files\dotnet\sdk]
  2.1.505 [C:\Program Files\dotnet\sdk]
  2.1.602 [C:\Program Files\dotnet\sdk]
  2.2.105 [C:\Program Files\dotnet\sdk]
  2.2.202 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

### PR DESCRIPTION
**Pull request by yufeih**
Fixes #4615

### CODE REVIEW

File: src/docfx/lib/git/FileCommitProvider.cs [hunk 1]
```diff
@@ -237,10 +238,11 @@ private unsafe (List<Commit>, Dictionary<long, Commit>) LoadCommits(string commi
                 // https://github.com/libgit2/libgit2sharp/issues/1351
                 if (error != 0 /* GIT_ENOTFOUND */)
                 {
-                    git_revwalk_free(walk);
-                    throw new NotSupportedException($"Error computing git log '{error}' for '{_repoPath}'");
```
Comment (OsmondJiang) [inline src/docfx/lib/git/FileCommitProvider.cs]: why directly throw exception doesn't break the build?

File: src/docfx/lib/git/FileCommitProvider.cs [hunk 2]
```diff
@@ -238,9 +239,12 @@ private unsafe (List<Commit>, Dictionary<long, Commit>) LoadCommits(string commi
                 if (error != 0 /* GIT_ENOTFOUND */)
```
Comment (live1206) [inline src/docfx/lib/git/FileCommitProvider.cs]: GIT_ENOTFOUND -> GIT_NOTFOUND

File: src/docfx/lib/git/FileCommitProvider.cs [hunk 3]
```diff
@@ -238,9 +239,12 @@ private unsafe (List<Commit>, Dictionary<long, Commit>) LoadCommits(string commi
...
                 {
                     git_revwalk_free(walk);
-                    throw new NotSupportedException($"Error computing git log '{error}' for '{_repoPath}'");
+
+                    Log.Write($"Load git commit failed: {error} {lastCommitId}");
+                    throw Errors.GitCloneIncomplete(_repoPath).ToException();
                 }
 
+                lastCommitId = commitId;
```
Comment (live1206) [inline src/docfx/lib/git/FileCommitProvider.cs]: what is this assignment for since it is not used any further?

Comment (yufeih) [reply src/docfx/lib/git/FileCommitProvider.cs]: used in `Log.Write`

File: src/docfx/lib/git/FileCommitProvider.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/lib/git/FileCommitProvider.cs]: That is the name defined in libgit2

File: src/docfx/lib/git/FileCommitProvider.cs (hunk 3 above)
Comment (live1206) [reply src/docfx/lib/git/FileCommitProvider.cs]: `Log.Write` is called before this assignment

Comment (live1206) [reply src/docfx/lib/git/FileCommitProvider.cs]: I see now

**Pull request merged**